### ISSUE TITLE
Add lesson for qualitative data analysis

### ISSUE THREAD

**Issue by gtlaflair** [datacarpentry/socialsci-workshop#19]
Would it be in our interests to add lessons for doing qualitative data analysis (of interview or conversation data that needs hand-coding for example) in R and Python.  I know of an R package that exists for this (RQDA) in which people can have/create a code book and uses a GUI within R to code data. Does something like this exist in Python? This is something that would probably add about half a day's worth of time.

**Comment by atheobold** [comment]
I agree with @gtlaflair that the research of many social scientists, myself included, necessitates qualitative data analyses. I was aware of the RQDA package, but had not put much time into researching its uses. After reading the link you provided, it appears that the package primarily uses keywords to define codes, which can then be organized into themes. Speaking from personal experience, I have had very few QDA projects where the analysis was as simple as finding all of the cases containing a specific keyword(s) and assigning them all to a single code. Instead, I holistically analyze text excerpts to assess whether they capture the same theme seen in the excerpts of other cases. 

These criticisms aside, I do think adding a brief lesson on the complexities of QDA would be interesting. Personally, I have only used paid QDA software (NVivio & MaxQDA), but think there is some free software that might better capture the complexities of QDA (e.g. Coding Analysis Toolkit, FreeQDA, etc.).

**Comment by gtlaflair** [comment]
@atheobold thanks for the reply! I think it might be the case that the link I provided doesn't give the deepest overview into RQDA's features. Here is [another resource](https://nsuworks.nova.edu/cgi/viewcontent.cgi?article=2659&context=tqr) that has a tutorial on its use for thematic analysis. I'm not sure if this expands to the use case you described, but I _think_ it's a step further than keyword analysis.

**Comment by atheobold** [comment]
@gtlaflair thanks for the more extensive documentation. Yes, it does appear that RQDA does have more QDA features than keyword searches! It does appear that the user is able to grab sections of text and assign them to codes, which can the be clustered into themes or categories. I did notice, however, when attempting to install that `RQDA` package, the package is not available for the current version of `R`. Perhaps this will change in the future, but it does make incorporation rather tricky. Additionally, I'm unsure what content could/should be cut from the current curriculum to allow for a 1/2 day lesson on QDA.

**Comment by gtlaflair** [comment]
You're welcome @atheobold! I agree, I'm not sure what could/should be cut to fit it into the lessons. Maybe it something to just continue to keep an eye on, especially if the package is unavailable for the most current version of `R`. It's too bad if it falls by the wayside. It did seem like a nice alternative to more costly QDA software packages. It seems like the type of package that the contributors at [rOpenSci](https://ropensci.org/) would have some relevant experience with, but I'm not sure how to go about making those introductions :)

**Comment by ndporter** [comment]
My experience with RQDA is it's very limiting, very few people use it, it's not being actively developed, and there's no parallel in Python. 

That said, I'll be developing a series of Carpentries-style lessons on Open Qualitative Research as part of an IMLS-funded initiative this fall that could also be adapted as either a separate lesson in the social science curriculum or a separate Data Carpentry curriculum. It will cover principles of qualitative coding, qualitative coding in spreadsheets, [Taguette](https://www.taguette.org/) (a FOSS qualitative coding tool) and some information on paid packages (what are major features of typical packages and how can you use open data formats like QDPX to share between them).
